interface/src/app/gene/searchMany/searchMany.js

```diff
@@ -36,226 +36,151 @@ angular.module('adage.gene.searchMany', [
 
-.factory('SearchResults', ['$rootScope', 'Gene', function($rootScope, Gene) {
-  var queries = [];
-  var searchResults = {};
-  var searchSuccess = null;
 
+// Directive for whole gene search form
+.directive('geneSearchPanel', [function() {
   return {
-    getQueries: function() {
-      return queries;
-    },
-    getQueryResults: function(query) {
-      return searchResults[query];
-    },
-    getSearchResults: function() {
-      return searchResults;
-    },
-    getSearchSuccess: function() {
-      return searchSuccess;
-    },
-    remove: function(query) { // Remove a query and its associated result
-      queries = queries.filter(function(el) {
-        return el !== query;
-      });
-      delete searchResults[query];
-      $rootScope.$broadcast('results.update');
-    },
-    clear: function() { // Clear the service
-      queries = [];
-      searchResults = {};
-      $rootScope.$broadcast('results.update');
-    },
-    size: function() {
-      return queries.length;
+    controller: ['$scope', function($scope) {
+      $scope.loadingSearchResults = false;
+
+      $scope.queries = [];
+      $scope.searchResults = {};
+    }],
+    restrict: 'E',
+    scope: {
+      organism: '='
     },
-    search: function(qparams) {
-      // Gene.search will query for all of the search terms in qparams
-      // returning a list of objects containing each search term and the
-      // gene results for that term in each objects. If the search
-      // contained terms already found in our cache of searchResults,
-      // those results are ignored while terms not already present will
-      // be added to the cache.
-      Gene.search(qparams,
-        function(data) {
-          var previousQueries = queries.length;
-          for (var i = 0; i < data.length; i++) {
-            var query = data[i].search;
-            if (!searchResults[query]) {
-              // If search term didn't already exist
-              searchResults[query] = data[i]; // add it to the results
-              queries.push(query); // add to the list of queries
-            }
-          }
-          if (previousQueries !== queries.length) {
-            $rootScope.$broadcast('results.update');
-          }
-          searchSuccess = true;
-          $rootScope.$broadcast('results.searchResultsReturned');
-        },
-        function(responseObject, responseHeaders) {
-          searchSuccess = false;
-          $rootScope.$broadcast('results.searchResultsReturned');
-        }
-      );
-    }
+    templateUrl: 'gene/searchMany/gene-search-panel.tpl.html'
   };
 }])
 
 
 // Directive for whole gene search form
-.directive('geneSearchForm', ['SearchResults', function(SearchResults) {
+.directive('geneSearchForm', [function() {
   return {
-    controller: ['$scope', '$rootScope', 'SearchResults',
-      function($scope, $rootScope, SearchResults) {
-        $scope.loadingSearchResults = false;
+    controller: ['$scope', 'Gene', function($scope, Gene) {
```

Comment:
It is worth considering whether all directives in this file should use link functions instead of controllers. (I didn't realize the difference until I wrote my own directive recently.)